src/core/models/dt_linearmodel.h

```diff
@@ -0,0 +1,181 @@
+//------------------------------------------------------------------------------
+// Copyright 2021 H2O.ai
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//------------------------------------------------------------------------------
+#ifndef dt_MODELS_LINEARMODEL_h
+#define dt_MODELS_LINEARMODEL_h
+#include "models/utils.h"
+#include "models/dt_linearmodel_base.h"
+#include "str/py_str.h"
+#include "models/label_encode.h"
+
+
+namespace dt {
+
+
+/**
+ *  Class template that implements all the virtual methods declared in
+ *  dt::LinearModelBase.
+ */
+template <typename T /* float or double */>
+class LinearModel : public dt::LinearModelBase {
+  private:
+    // SType that corresponds to `T`
+    const SType stype;
+    size_t : 56;
+    // Model's shape is (nfeatures + 1, nlabels)
+    dtptr dt_model;
+    std::vector<T*> w;
```

Comment:
Each element of this vector corresponds to the model coefficients for one particular label. The size of each vector element is `nfeatures + 1`, because we also have an intercept term. Not sure why the size should be different in the multinomial case? Multinomial classification here ends up as a set of binomial classifiers.

Reply:
ah yes, if the intercept is implicit then one more parameter is needed. 
For multinomial, one of the coefficients is always set to 0 because otherwise they are linearly dependent: https://en.wikipedia.org/wiki/Multinomial_logistic_regression#As_a_set_of_independent_binary_regressions

Reply:
TODO: check if we want to implement multinomial logistic regression